cartography/intel/gcp/bigtable.py

```
@@ -0,0 +1,350 @@
+import logging
+from typing import Any
+from typing import Dict
+from typing import List
+
+import neo4j
+from google.api_core.exceptions import PermissionDenied
+from google.auth.exceptions import DefaultCredentialsError
+from google.auth.exceptions import RefreshError
+from googleapiclient.discovery import HttpError
+from googleapiclient.discovery import Resource
+
+from cartography.client.core.tx import load
+from cartography.graph.job import GraphJob
+from cartography.models.gcp.bigtable.app_profile import GCPBigtableAppProfileSchema
+from cartography.models.gcp.bigtable.backup import GCPBigtableBackupSchema
+from cartography.models.gcp.bigtable.cluster import GCPBigtableClusterSchema
+from cartography.models.gcp.bigtable.instance import GCPBigtableInstanceSchema
+from cartography.models.gcp.bigtable.table import GCPBigtableTableSchema
+from cartography.util import timeit
+
+logger = logging.getLogger(__name__)
+
+
+@timeit
+def get_bigtable_instances(client: Resource, project_id: str) -> List[Dict]:
+    instances: List[Dict] = []
+    try:
```

no need for defensive programming in this function unless you are seeing these errors when you are testing your live sync. Better to let it fail loudly